src/components/AnchorBase.tsx

```diff
@@ -7,24 +7,34 @@ import { isSidenoteSelected } from '../store/ui/selectors';
 import { Dispatch, State } from '../store';
 import { getDoc } from './utils';
 
+
+/**
+ * AnchorBase Props
+ */
 type Props = {
   anchor: string;
   className?: string;
   children: React.ReactNode;
+  /** when true (default) the component will disconnect from the store on unmount*/
+  autoDisconnect?: boolean;
 };
 
+
 export const AnchorBase = (props: Props) => {
   const {
-    anchor, children, className,
+    anchor, children, className, autoDisconnect,
   } = props;
   const dispatch = useDispatch<Dispatch>();
   const [doc, setDoc] = useState<string>();
   const [ref, setRef] = useState<HTMLDivElement | null>(null);
 
   useEffect(() => {
-    if (ref == null || doc == null) return () => {};
-    return () => dispatch(disconnectAnchor(doc, ref));
-  }, [doc, ref]);
+    if (!ref || !autoDisconnect) return () => null;
+    return () => {
+      const parentDoc = doc ?? getDoc(ref);
```

Comment:
This seems to cover the use case you removed below. In one action. So should be good.